src/embedded.js

```diff
@@ -1,981 +1,1051 @@
-                        url = url.replace(/\&amp\;/g, '&');
-                    }
-                    else {
-                        url = decodedUrl;
-                    }
-                }
-                catch (e) {
-                    l('Could not decode url: ' + e);
-                }
-            }
-            return url;
-        }
-    };
-
-    /**
-     * Wrapper that will ensure an error message is displayed, either in console.log
-     * or as a browser alert.
-     * @param message String error message
-     */
-
-    function error(message) {
-        if (typeof message !== 'undefined') {
-            if (window.console && console.log) {
-                console.log(message);
-            } else {
-                alert(message);
-            }
-        }
+    if (typeof val !== 'string') {
+      throw new TypeError('"clientId" must be a string');
     }
 
-    /**
-     * Custom wrapper that conditionally logs messages to console.log.
-     * @param messageObj String or Object to log
-     */
-    function l(messageObj) {
-        if (HelloSign.isDebugEnabled && typeof messageObj !== 'undefined' &&
-            window.console && console.log) {
-            console.log(messageObj);
-        }
+    params.append('client_id', val);
+  }
+
+  /**
+   * Validates and appends the debug parameter to the
+   * iFrame params object.
+   *
+   * @throws {TypeError} if debug is invalid
+   * @param {URLSearchParams} params
+   * @private
+   */
+  _applyDebug(params) {
+    const val = this._config.debug;
+
+    if (typeof val !== 'boolean') {
+      throw new TypeError('"debug" must be a boolean');
     }
 
-    /**
-     *  Getter functions for determining scroll position that work on all
-     *  browsers.
-     */
+    params.append('debug', val ? 1 : 0);
+  }
+
+  /**
+   * Validates and appends the finalButtonText parameter to
+   * the iFrame params object.
+   *
+   * @throws {TypeError} if finalButtonText is invalid
+   * @param {URLSearchParams} params
+   * @private
+   */
+  _applyFinalButtonText(params) {
+    if ('finalButtonText' in this._config) {
+      const val = this._config.finalButtonText;
+
+      if (typeof val !== 'string') {
+        throw new TypeError('"finalButtonText" must be a string');
+      }
+
+      if (!['Send', 'Continue'].includes(val)) {
+        throw new TypeError('"finalButtonText" must be either "Send" or "Continue"');
+      }
+
+      params.append('final_button_text', val);
+    }
+  }
+
+  /**
+   * Validates and appends the hideHeader parameter to the
+   * iFrame params object.
+   *
+   * @throws {TypeError} if hideHeader is invalid
+   * @param {URLSearchParams} params
+   * @private
+   */
+  _applyHideHeader(params) {
+    if ('hideHeader' in this._config) {
+      const val = this._config.hideHeader;
+
+      if (typeof val !== 'boolean') {
+        throw new TypeError('"hideHeader" must be a boolean');
+      }
+
+      params.append('hideHeader', val);
```

Comment:
Why is this param camel case? All the other params are snake case.

Reply:
Yeah I know, it's just how the endpoint was written. Out of my control :(

Reply:
Why don't we just update the endpoint to support both? Would be one line of code!

Reply:
But good callout, because it does look like it's unintentional

Reply:
I just think v2 is a good opportunity to clean stuff like this up